### DISCUSSION TITLE
macbook-air-m2-perfect-device-developers/

### DISCUSSION THREAD

**Discussion by giscus** [estruyf/eliostruyf-comments#614] [Comments]
# macbook-air-m2-perfect-device-developers/

Azure / Microsoft 365 / Visual Studio Code / Speaker / Blogger

https://www.eliostruyf.com/macbook-air-m2-perfect-device-developers/

**Comment by pacmansp** [comment]
Good write up. I'm actually moving into web development in the next year or so at work. Focusing on Ruby on Rails at first, since that's what my job currently uses, then will branch out from there. I'm looking to purchase a laptop that can be my main device for the next few years. Would you recommend the M2 MBA if it is your only device for web development? I'd be getting the 16GB 512GB version if I buy it.

I'm a big fan of the new 15" version when I've played with it in stores.

**Reply by estruyf** [reply]
I am still happy with the device. If you are going to use it for web development, it would be a great fit.

**Comment by aravinds13** [comment]
Thanks for the detailed review! I'm in the market for a new MacBook that I will be using primarily for university (CS student) and some web and mobile app development. I've been considering the M2 MBA and the M2 MBP. Do you think the M2 MBP would have a significant performance bump of the M2 MBA because of active cooling?

**Reply by estruyf** [reply]
Hi @aravinds13, I like the MBA. I can do all of the things without any performance issues. At home, as mentioned, my primary device is an Apple Mini M2, but doing the same tasks on it apart from some video recording and editing. If that would be something you are going to do as well, I would go for the MBP option; otherwise, the MBA is light, silence, slim, ...

**Comment by markmaker** [comment]
Thank you, very useful. Have you ever done much C++ compiling, running Docker with ARM Linux VM or other heavy lifting?
